scripts/check_names.py

```diff
@@ -771,64 +804,180 @@ def parse_symbols(self):
 
             # Perform object file analysis using nm
             symbols = self.parse_symbols_from_nm([
-                "library/libmbedcrypto.a",
-                "library/libmbedtls.a",
-                "library/libmbedx509.a"
+                self.out_of_source_dir + "/drivers/builtin/libbuiltin.a",
+                self.out_of_source_dir + "/drivers/p256-m/libp256m.a",
+                self.out_of_source_dir + "/drivers/everest/libeverest.a",
+                self.out_of_source_dir + "/core/libtfpsacrypto.a"
             ])
 
-            subprocess.run(
-                ["make", "clean"],
-                universal_newlines=True,
-                check=True
-            )
+            os.chdir(previous_dir)
         except subprocess.CalledProcessError as error:
             self.log.debug(error.output)
             raise error
         finally:
             # Put back the original config regardless of there being errors.
             # Works also for keyboard interrupts.
             shutil.move(
-                "include/mbedtls/mbedtls_config.h.bak",
-                "include/mbedtls/mbedtls_config.h"
+                self.source_dir + "/include/psa/crypto_config.h.bak",
+                self.source_dir + "/include/psa/crypto_config.h"
             )
 
         return symbols
 
-    def parse_symbols_from_nm(self, object_files):
+class MBEDTLSCodeParser(CodeParser):
+    """
+    Class for retrieving files and parsing Mbed TLS code. This can be used
+    independently of the checks that NameChecker performs, for example for
+    list_internal_identifiers.py.
```

Comment:
```suggestion
    Class for retrieving files and parsing Mbed TLS code. This can be used
    independently of the checks that NameChecker performs.
```
ditto